staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -695,6 +695,10 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 			break
 		}
 		key = string(lastKey) + "\x00"
+		if fromRV == nil {
+			*fromRV = uint64(returnedRV)
+			options = append(options, clientv3.WithRev(returnedRV))
```

Comment:
appending to options inside this for loop looks wrong... that means we'll have multiple WithRev options specified with different values

Reply:
edit: nevermind, I see this only happens when fromRV is nil